rflib/main/default/lwc/rflibLogEventMonitor/__tests__/rflibLogEventMonitor.test.js

```diff
@@ -0,0 +1,285 @@
+    return {
+        loadStyle: jest.fn()
+    };
+});
+
+// Mock Apex
+jest.mock(
+    '@salesforce/apex/rflib_LogArchiveController.getArchivedRecords',
+    () => {
+        return {
+            default: jest.fn()
+        };
+    },
+    { virtual: true }
+);
+
+jest.mock(
+    '@salesforce/apex/rflib_LogArchiveController.clearArchive',
+    () => {
+        return {
+            default: jest.fn()
+        };
+    },
+    { virtual: true }
+);
+
+// Mock resourceUrl
+jest.mock(
+    '@salesforce/resourceUrl/rflib_HidePageHeader',
+    () => {
+        return 'rflib_HidePageHeader';
+    },
+    { virtual: true }
+);
+
+// Helper to flush promises
+function flushPromises() {
+    return new Promise(resolve => setTimeout(resolve, 0));
+}
+
+describe('c-rflib-log-event-monitor', () => {
+    afterEach(() => {
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+        jest.clearAllMocks();
+    });
+
+    it('subscribes to event channel on initialization', () => {
+        const element = createElement('c-rflib-log-event-monitor', {
+            is: RflibLogEventMonitor
+        });
+
+        isEmpEnabled.mockResolvedValue(true);
+        subscribe.mockResolvedValue({ channel: '/event/rflib_Log_Event__e' });
+        loadStyle.mockResolvedValue();
+
+        document.body.appendChild(element);
+
+        return flushPromises().then(() => {
+            expect(isEmpEnabled).toHaveBeenCalled();
+            expect(subscribe).toHaveBeenCalled();
+            expect(subscribe.mock.calls[0][0]).toBe('/event/rflib_Log_Event__e');
+        });
+    });
+
+    it('handles new messages from subscription', () => {
+        const element = createElement('c-rflib-log-event-monitor', {
+            is: RflibLogEventMonitor
+        });
+
+        isEmpEnabled.mockResolvedValue(true);
+        let messageCallback;
+        subscribe.mockImplementation((channel, replayId, callback) => {
+            messageCallback = callback;
+            return Promise.resolve({ channel });
+        });
+        loadStyle.mockResolvedValue();
+
+        document.body.appendChild(element);
+
+        return flushPromises()
+            .then(() => {
+                const message = {
+                    data: {
+                        payload: {
+                            CreatedById: 'User1',
+                            Log_Level__c: 'ERROR'
+                        }
+                    }
+                };
+                if(messageCallback) messageCallback(message);
+                return flushPromises();
+            })
+            .then(() => {
+                const logEventList = element.shadowRoot.querySelector('c-rflib-log-event-list');
+                expect(logEventList.logEvents).toHaveLength(1);
+                expect(logEventList.logEvents[0].CreatedById).toBe('User1');
+            });
+    });
+
+    it('handles connection mode change to Archive', () => {
```

Comment:
Add more test cases that handle the switching of connection modes in the context of managing the subscription to the event channel.